Unexpected dsd in read_message

This is an error I receive
```
        # Instantiate reader
        reader = Reader()
    
        # Parse the message, using any provided or auto-queried DSD
        msg = reader.read_message(response_content,
>                                 dsd=dsd)
E       TypeError: read_message() got an unexpected keyword argument 'dsd'

../../../../../opt/miniconda3/envs/pmpx/lib/python3.7/site-packages/pandasdmx/api.py:423: TypeError
```
I think the problem is that BaseReader and json reader both do not expect "dsd" at all. Only xml reader has this dsd thing in the `read_message` method. This is kind of violation of inheritance principle from abstract - i.e. method definitions should be the same as abstract. 


Although, you are always passing it in the api:

![image](https://user-images.githubusercontent.com/41692261/82908707-41552f80-9f60-11ea-94ad-f9848d63660e.png)

In this case I am querying OECD where I believe default format is json.

--- a/README.rst
+++ b/README.rst
@@ -30,7 +30,7 @@ central banks, and international organisations.
 
 - explore the data available from `data providers
   <https://pandasdmx.readthedocs.io/en/latest/sources.html>`_
-  such as the World Bank, International Monetary Fund, Eurostat, OECD, and United Nations;
+  such as the World Bank, ILO, ECB, Eurostat, OECD, and United Nations;
 - parse data and metadata in SDMX-ML (XML) or SDMX-JSON formats—either:
 
   - from local files, or

--- a/doc/index.rst
+++ b/doc/index.rst
@@ -8,7 +8,8 @@ agencies, central banks, and international organisations.
 
 :mod:`pandaSDMX` can be used to:
 
-- explore the data available from :doc:`data providers <sources>` such as the World Bank, International Monetary Fund, Eurostat, OECD, and United Nations;
+- explore the data available from :doc:`data providers <sources>` such as the World Bank, ILO, ECB,  
+  Eurostat, OECD, and United Nations;
 - parse data and metadata in SDMX-ML (XML) or SDMX-JSON formats—either:
 
   - from local files, or

--- a/doc/sources.rst
+++ b/doc/sources.rst
@@ -139,8 +139,9 @@ SDMX-ML —
 SDMX-ML —
 `Website <https://sdmxcentral.imf.org/>`__
 
-- Subset of the data available on http://data.imf.org.
-- Supports series-key-only and hence dataset-based key validation and construction.
+- The SDMX REST API no longer accepts data queries. But queries for metadata still work.
+  Datasets must be retrieved manually from 
+  https://data.imf.org.  
 
 
 ``INEGI``: National Institute of Statistics and Geography (Mexico)

--- a/doc/whatsnew.rst
+++ b/doc/whatsnew.rst
@@ -3,6 +3,15 @@
 What's new?
 ===========
 
+
+
+v1.0.1 (2020-05-28)
+-------------------------------
+
+- IMF no longer accepts data queries. Update source and docs to reflect this.
+- Fix crash when making data requests to JSON-based data sources
+
+
 v1.0.0 (2020-05-15)
 -------------------------------
 

--- a/pandasdmx/__init__.py
+++ b/pandasdmx/__init__.py
@@ -15,7 +15,7 @@ __all__ = [
     ]
 
 
-__version__ = '1.0.0'
+__version__ = '1.0.1'
 
 
 #: Top-level logger for pandaSDMX.

--- a/pandasdmx/api.py
+++ b/pandasdmx/api.py
@@ -418,8 +418,12 @@ class Request:
         # Instantiate reader
         reader = Reader()
 
-        # Parse the message, using any provided or auto-queried DSD
-        msg = reader.read_message(response_content,
+        # Parse the message 
+        # Only if   SDMXML content is given, use any provided or auto-queried DSD.
+        if 'json' in content_type:
+            msg = reader.read_message(response_content)
+        else:
+            msg = reader.read_message(response_content,
                                   dsd=dsd)
 
         # Store the HTTP response with the message

--- a/pandasdmx/sources.json
+++ b/pandasdmx/sources.json
@@ -45,6 +45,7 @@
     "url": "https://sdmxcentral.imf.org/ws/public/sdmxapi/rest",
     "name": "International Monetary Fund",
     "supports": {
+      "data": false,
       "provisionagreement": false
     }
   },
@@ -113,7 +114,10 @@
     "id": "UNSD",
     "name": "United Nations Statistics Division",
     "url": "http://data.un.org/WS/rest",
-    "supports": {"preview": true, "provisionagreement": false}
+    "supports": {
+      "preview": true, 
+      "provisionagreement": false
+    }
   },
   {
     "id": "WB",

